### DISCUSSION TITLE
Actions are undefined in tests

### DISCUSSION THREAD

**Discussion by rgrosjean** [vuejs/pinia#2464] [Help and Questions] [answered]
Hello everyone,

I'm not sure I'm in the right section for this, but I'm still not sure if it's a bug or just an obvious mistake from my side.

I'm trying to test a component where an action is a store is used in a writable computed
It looks like this
```
const userStore = useUserStore();

const firstName = computed({
  get() {
    return userStore.imsUser ? userStore.imsUser.firstName : null;
  },
  set(firstName) {
    userStore.setFirstName(firstName)
  }
});
```

In my test, I think that I have done everything as needed, but I always get this error:

`TypeError: userStore.setFirstName is not a function`

```
import { describe, it, expect, vi } from "vitest"
import { createTestingPinia } from "@pinia/testing"
import { mount } from "@vue/test-utils"
import { useUserStore } from "@/stores/user"

describe("MyModule.vue", () => {

  const wrapper = mount(EditNameVue, {
    global: {
      plugins: [
        createTestingPinia({
          createSpy: vi.fn(),
          initialState: {
            user: {
              // initial state
            }
          }
        })
      ]
    }
  })
  const userStore = useUserStore()
  it("does something", () => {
      const inputs = wrapper.findAllComponents(LmTextField)
      const firstName = inputs[0]

      firstName.setValue("change")
    })
}
```

After some debugging, it seems that all the actions of my userStore are undefined. The value of  `stubActions` does not have any effect.

What am I doing wrong?

- @pinia/testing@0.1.3
- @vue/tsconfig@0.4.0
- pinia@2.1.6
- vue@3.3.4

**Comment by posva** [comment, marked as answer]
Your `createSpy` option is not correct See https://pinia.vuejs.org/cookbook/testing.html#Testing-stores:


```ts
createSpy: vi.fn,
```

I improved the docs at 40f8f22

**Reply by rgrosjean** [reply]
I've said it's something stupid. Thanks a lot!
